### ISSUE TITLE
Incorrect comment in security documentation

### ISSUE THREAD

**Issue by pathmissing** [symfony/symfony-docs#8535]
In the security documentation under "Retrieving the User Object" in the code example, there is a comment saying "the above is a shortcut for this".

```php
use Symfony\Component\Security\Core\User\UserInterface;

public function indexAction(UserInterface $user)
{
    if (!$this->get('security.authorization_checker')->isGranted('IS_AUTHENTICATED_FULLY')) {
        throw $this->createAccessDeniedException();
    }

    // the above is a shortcut for this
    $user = $this->get('security.token_storage')->getToken()->getUser();
}
```

Which is incorrect. Or am I missing something?

**Comment by weaverryan** [comment]
Ah, yea, it’s very unclear - the above is referring to the User argument to the controller I think.. so that should indeed be fixed - very confusing!

**Comment by pathmissing** [comment]
I actually did not even look at the method signature to be honest. Now it just looks self explanatory. But mentioning the possibility of type hinting the UserInterface in the description above the code example would indeed clarify this even more. Thank you and everyone else for the great work with the documentation!

**Comment by xabbuh** [comment]
@pathmissing Thanks for reporting this. Would you like to open a pull request to improve this example? :)

**Comment by javiereguiluz** [comment]
Closing as fixed by #8542.
